### PR TITLE
AddTemplate to ES in grill

### PR DESCRIPTION
**Pull request by nileshkadamawg**
API to add template to ES via grill

### CODE REVIEW

File: pkg/grillelasticsearch/stubs.go [hunk 1]
```diff
@@ -49,3 +49,23 @@ func (ge *ElasticSearch) UpsertItem(index, docId, data string) grill.Stub {
...
+func (ge *ElasticSearch) AddTemplate(name string, data string) grill.Stub {
+	return grill.StubFunc(func() error {
+		req := esapi.IndicesPutTemplateRequest{
+			Body: strings.NewReader(data),
+			Name: name,
+		}
+		res, err := req.Do(context.Background(), ge.elasticSearch.Client)
+		if err != nil {
+			return err
+		}
```
Comment (amit-shekhar) [inline pkg/grillelasticsearch/stubs.go:62]: add defer res.Body.Close()

File: pkg/grillelasticsearch/stubs.go [hunk 2]
```diff
@@ -49,3 +49,23 @@ func (ge *ElasticSearch) UpsertItem(index, docId, data string) grill.Stub {
 		return nil
 	})
 }
+
+func (ge *ElasticSearch) AddTemplate(name string, data string) grill.Stub {
```
Comment (amit-shekhar) [inline pkg/grillelasticsearch/stubs.go]: variable name -> data to template

**Review by amit-shekhar — CHANGES_REQUESTED** [review]

File: pkg/grillelasticsearch/stubs.go [hunk 3]
```diff
@@ -49,3 +49,24 @@ func (ge *ElasticSearch) UpsertItem(index, docId, data string) grill.Stub {
...
 }
+
+func (ge *ElasticSearch) AddScript(name string, script string) grill.Stub {
+	return grill.StubFunc(func() error {
+		req := esapi.PutScriptRequest{
+			Body:       strings.NewReader(script),
+			DocumentID: name,
+		}
+		res, err := req.Do(context.Background(), ge.elasticSearch.Client)
+		defer res.Body.Close()
```
Comment (amit-shekhar) [inline pkg/grillelasticsearch/stubs.go]: move this after the error check. in case of error, res can be nil

Comment (nileshkadamawg) [reply pkg/grillelasticsearch/stubs.go]: ok

File: pkg/grillelasticsearch/assertions.go [hunk 4]
```diff
@@ -51,3 +52,24 @@ func (ge *ElasticSearch) AssertItemsCount(index string, count int, expected []js
 		return nil
 	})
 }
+
+func (ge *ElasticSearch) AssertScript(name string) grill.Assertion {
```
Comment (amit-shekhar) [inline pkg/grillelasticsearch/assertions.go]: what is the purpose of assertscript ?? 

do we create and remove scripts in runtime, or is it a one time thing at the beginning ?

Comment (nileshkadamawg) [reply pkg/grillelasticsearch/assertions.go]: We are creating and deleting also in run time.

**Review by amit-shekhar — APPROVED** [review]

**Pull request merged**